types/jasmine/index.d.ts

```diff
@@ -858,14 +858,60 @@ declare namespace jasmine {
     interface PassedExpectation extends CustomReportExpectation {
     }
 
+    interface DeprecatedExpectation {
+        message: string;
+    }
+
     interface CustomReporterResult {
+        /**
+         * The unique id of this spec.
+         */
+        id: string;
+
+        /**
+         * The description passed to the {@link it} that created this spec.
+         */
         description: string;
-        failedExpectations?: FailedExpectation[];
+
+        /**
+         * The full description including all ancestors of this spec.
+         */
         fullName: string;
-        id: string;
+
+        /**
+         * The list of expectations that failed during execution of this spec.
+         */
+        failedExpectations?: FailedExpectation[];
+
+        /**
+         * The list of expectations that passed during execution of this spec.
+         */
         passedExpectations?: PassedExpectation[];
+
+        /**
+         * The list of deprecation warnings that occurred during execution this spec.
+         */
+        deprecationWarnings?: DeprecatedExpectation[];
+
+        /**
+         * If the spec is pending, this will be the reason.
+         */
         pendingReason?: string;
+
+        /**
+         * Once the spec has completed, this string represents the pass/fail status of this spec.
+         */
         status?: string;
+
+        /**
+         * The time in ms used by the spec execution, including any before/afterEach.
+         */
+        duration: number | null;
```

Comment:
I see that this + L914 are typed as possible nulls, because the Jasmine code initializes this.result to have those fields set to null.
But in practice, are they always set? Because by the time the spec finishes executing, won't those values exist then?

Reply:
It's always set as a number in specDone(), but it's passed as null to specStarted().

These types could be better if we split this interface into a different type for each of {specStarted,specDone,suiteStarted,suiteDone}, but I didn't want to change things that much in this PR.